src/annotation.js

```diff
@@ -29,58 +29,71 @@ export default {
   beforeInit(chart) {
     chartStates.set(chart, {
       elements: [],
+      options: {},
       listeners: {},
       listened: false,
       moveListened: false,
       scales: new Set()
     });
   },
 
-  afterDataLimits(chart, args, options) {
+  afterDataLimits(chart, args) {
     if (args.scale.type !== 'category') {
-      adjustScaleRange(chart, args.scale, options);
+      const state = chartStates.get(chart);
+      adjustScaleRange(args.scale, state.options);
     }
   },
 
   beforeUpdate(chart, args, options) {
-    if (isObject(options.annotations)) {
+    const state = chartStates.get(chart);
+    state.options = clone(options);
```

Comment:
@kurkle ok but it's not clear to me why we didn't use it before. For instance, for display we created another options in the element instead of change the options. I assumed you didn't want to change the options passed as argument. If My assumption is wrong, I can remove it

Reply:
Nvm, I was thinking we would only have the `annotations` array in state and would populate that with the merged options. We can change that later, if needed.

Reply:
> I was thinking we would only have the `annotations` array in state and would populate that with the merged options.

There is no vantages to have only the annotations because many methods are not not using only the "annotations" but whole plugin options object (see events). But let me try, maybe could make sense